test-network-function/networking/suite_test.go

```diff
@@ -0,0 +1,161 @@
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package networking
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/test-network-function/test-network-function/pkg/tnf/interactive"
+)
+
+func TestParseVariables(t *testing.T) {
+	// expected inputs
+	testCases := []struct {
+		// inputs
+		inputRes      string
+		declaredPorts map[key]string
+
+		// expected outputs here
+		expectedDeclaredPorts map[key]string
+		expectedRes           string
+	}{
+		{
+			inputRes:              `[\n  {\n    \"containerPort\": 8080,\n    \"name\": \"http-probe\",\n    \"protocol\": \"UDP\"\n  }\n]`,
+			declaredPorts:         map[key]string{},
+			expectedDeclaredPorts: map[key]string{{port: 8080, protocol: "UDP"}: "http-probe"},
+			expectedRes:           `[\n  {\n    \"containerPort\": 8080,\n    \"name\": \"http-probe\",\n    \"protocol\": \"UDP\"\n  }\n]`,
+		},
+		{
+			inputRes:              "",
+			declaredPorts:         map[key]string{},
+			expectedDeclaredPorts: map[key]string{},
+			expectedRes:           "",
+		},
+		{
+			inputRes:              `[\n  {\n    \"containerPort\": 8080,\n    \"name\": \"http-probe\",\n    \"protocol\": \"TCP\"\n  }\n]`,
+			declaredPorts:         map[key]string{},
+			expectedDeclaredPorts: map[key]string{{port: 8080, protocol: "TCP"}: "http-probe"},
+			expectedRes:           `[\n  {\n    \"containerPort\": 8080,\n    \"name\": \"http-probe\",\n    \"protocol\": \"TCP\"\n  }\n]`,
+		},
+	}
+
+	for _, tc := range testCases {
+		err := parseVariables(tc.inputRes, tc.declaredPorts)
+		assert.Nil(t, err)
+		assert.Equal(t, tc.expectedDeclaredPorts, tc.declaredPorts)
+	}
+}
+
+func TestDeclaredPortList(t *testing.T) {
+	// expected inputs
+	testCases := []struct {
+		// inputs
+		container     int
+		podName       string
+		podNamespace  string
+		declaredPorts map[key]string
+
+		// expected outputs here
+		expectedDeclaredPorts map[key]string
+	}{
+		{
+			container:             0,
+			podName:               "test-54bc4c6d7-2gwlz",
+			podNamespace:          "tnf",
+			declaredPorts:         map[key]string{},
+			expectedDeclaredPorts: map[key]string{{port: 8080, protocol: "TCP"}: "http-probe"},
+		},
+		{
+			container:             0,
+			podName:               "",
+			podNamespace:          "",
+			declaredPorts:         map[key]string{},
+			expectedDeclaredPorts: map[key]string{},
+		},
+	}
+
+	for _, tc := range testCases {
+		err := declaredPortList(tc.container, tc.podName, tc.podNamespace, tc.declaredPorts)
```

Comment:
This UT won't work unless you mock, for each test case, the utils.ExecuteCommand() function that declarePortList() is calling internally. Other than that, I don't see the point of having this UT, apart from testing a possible error thrown by the utils.ExecuteCommand() when parsing the output.

Reply:
I fixed it and this works now.